spec.emu

```diff
@@ -8,92 +8,260 @@ title: Await Dictionary
+        <emu-note>
+          <p>This function requires its *this* value to be a constructor function that supports the parameter conventions of the Promise constructor.</p>
+        </emu-note>
+
+        <emu-clause id="sec-performpromiseallkeyed" type="abstract operation">
+          <h1>
+            <ins>
+              PerformPromiseAllKeyed (
+                _promises_: an Object,
+                _constructor_: a constructor,
+                _resultCapability_: a PromiseCapability Record,
+                _promiseResolve_: a function object,
+              ): either a normal completion containing an ECMAScript language value or a throw completion
+            </ins>
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Let _allKeys_ be ? _promises_.[[OwnPropertyKeys]]().
+            1. Let _keys_ be a new empty List.
+            1. Let _values_ be a new empty List.
+            1. Let _remainingElementsCount_ be the Record { [[Value]]: 1 }.
+            1. Let _index_ be 0.
+            1. For each element _key_ of _allKeys_, do
+              1. Let _desc_ be ? _promises_.[[GetOwnProperty]](_key_).
+              1. If _desc_ is not *undefined* and _desc_.[[Enumerable]] is *true*, then
+                1. Let _value_ be ? Get(_promises_, _key_).
+                1. Append _key_ to _keys_.
+                1. Append *undefined* to _values_.
+                1. Let _nextPromise_ be ? Call(_promiseResolve_, _constructor_, « _value_ »).
+                1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-promise.allkeyed-resolve-element-functions" title></emu-xref>.
+                1. Let _length_ be the number of non-optional parameters of the function definition in <emu-xref href="#sec-promise.allkeyed-resolve-element-functions" title></emu-xref>.
+                1. Let _onFulfilled_ be CreateBuiltinFunction(_steps_, _length_, *""*, « [[AlreadyCalled]], [[Index]], [[Keys]], [[Values]], [[Capability]], [[RemainingElements]] »).
+                1. Set _onFulfilled_.[[AlreadyCalled]] to *false*.
+                1. Set _onFulfilled_.[[Index]] to _index_.
+                1. Set _onFulfilled_.[[Keys]] to _keys_.
+                1. Set _onFulfilled_.[[Values]] to _values_.
+                1. Set _onFulfilled_.[[Capability]] to _resultCapability_.
+                1. Set _onFulfilled_.[[RemainingElements]] to _remainingElementsCount_.
+                1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] + 1.
+                1. Perform ? Invoke(_nextPromise_, *"then"*, « _onFulfilled_, _resultCapability_.[[Reject]] »).
+                1. Set _index_ to _index_ + 1.
+            1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
+            1. If _remainingElementsCount_.[[Value]] = 0, then
+              1. NOTE: There may be non-zero results if a ill-behaved thenable synchronously invoked its then callback.
```

Comment:
Nit: this wording is strange to read after `_remainingElementsCount_.[[Value]] = 0`. Maybe instead

```suggestion
              1. NOTE: This can happen even if _keys_ was non-empty if an ill-behaved thenable synchronously invoked the callback passed to its *"then"* method.
```